pkg/api/folder.go

```diff
@@ -186,7 +198,7 @@ func (hs *HTTPServer) setDefaultFolderPermissions(ctx context.Context, orgID int
 			})
 		}
 
-		if !isNested {
+		if !isNested || !hs.Features.IsEnabled(featuremgmt.FlagNestedFolders) {
```

Comment:
Small change to make sure that we don't set permissions wrongly if parent folder for some reason is set while nested folders are not enabled.